opnode/rollup/driver/step.go

```diff
@@ -36,6 +38,54 @@ type outputImpl struct {
 	Config rollup.Config
 }
 
+func (d *outputImpl) newBlock(ctx context.Context, l2Finalized eth.BlockID, l2Parent eth.BlockID, l1Origin eth.BlockID, includeDeposits bool) (eth.BlockID, *derive.BatchData, error) {
+	d.log.Info("creating new block", "l2Parent", l2Parent, "l1Origin", l1Origin, "includeDeposits", includeDeposits)
+	fetchCtx, cancel := context.WithTimeout(ctx, time.Second*20)
+	defer cancel()
+	l2Info, err := d.rpc.BlockByHash(fetchCtx, l2Parent.Hash)
```

Comment:
We can check if the previous block has the same L1 information to determine if we've included deposits yet, and then we don't need the `includeDeposits` argument

Reply:
Doesn't work against the origin block

Reply:
We can make the l1-info parsing function just return an artificial l1 info, because the genesis block cannot have transactions. Not a big deal for this PR though, I'll open an issue to try reduce the function arguments here.